asyn/drvAsynFTDI/ftdiDriver.cpp

```diff
@@ -455,23 +560,48 @@ FTDIDriverStatus FTDIDriver::write(const unsigned char *buffer, int bufferSize,
   struct ftdi_transfer_control *tc = ftdi_write_data_submit (
     ftdi_, (unsigned char *) buffer, (int) bufferSize
   );
-  rc = ftdi_transfer_data_done (tc);
+  rc[8] = ftdi_transfer_data_done (tc);
 #else
-  rc = ftdi_write_data(ftdi_, (unsigned char *) buffer, (int) bufferSize);
+  if( spi ) {
+    if( !spiInit ) initSPI();
+    i = 0;
+    pbuf = (unsigned char *)&buf[0];
+
+    memcpy((pbuf+i), buffer, bufferSize);
+    i+= bufferSize;
+
+    debugPrint("Preamble+buffer:");
+    for (int j = 0; j < i; j++) debugPrint("=[%02X] ", *(pbuf+j));
+    debugPrint("\n");
+
+    flush();
+    rc[3] = ftdi_write_data(ftdi_, pbuf, i);
+    if( rc[3] != i ) err = FTDIDriverError;
+
+    } // if SPI -
+  else {
+    rc[7] = ftdi_write_data(ftdi_, (unsigned char *) buffer, (int) bufferSize);
+    if( rc[7] != i ) err = FTDIDriverError;
+    debugPrint("%s : 7. pinState=0x%02x\n",
+ 		functionName, pinState);
+    }
 #endif
 
   gettimeofday(&ctime, NULL);
   tnow = ((ctime.tv_sec - stime.tv_sec) * 1000) + (ctime.tv_usec / 1000);
   debugPrint("%s : Time taken for write => %ld ms\n", functionName, (tnow-(mtimeout-timeout)));
-  if (rc > 0){
-    debugPrint("%s : %d bytes written\n", functionName, rc);
-    *bytesWritten = (size_t)rc;
+  if (rc[3] > 0){
```

Comment:
It seems that throughout this function only `rc[3]` and `rc[7]` are ever used, and disjointly. Maybe I am missing the reason for using an array of return codes, but I believe it would be better to change these instances to simply use `rc` (a regular, singular `size_t`).